### ISSUE TITLE
use Singleton design pattern in ce_main_app

### ISSUE THREAD

**Issue by miniupnp** [atarijookie/ce-atari#143]
a lot of object in ce_main_app are Signeton (object that must have one instance)
access to theses objects are not very clean : through SharedObjects or local pointer to the single instance, etc.

**Comment by mikrosk** [comment]
Yeah, I've seen a lot of non-`volatile` variables and I could only hope that due to the fact they're `external` the compiler wont cache their values. Plus as you say, it's not always very safely done, somewhere are mutexes, somewhere they are not...

**Comment by miniupnp** [comment]
see also 2684c07f2af13c49004828fc0c433bce697810e8
